hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationRunner.java

```diff
@@ -170,6 +171,9 @@ public static void runDirect(Operation operation) throws Exception {
             operation.afterRun();
         } finally {
             operation.popThreadContext();
+            if (operation instanceof MapOperation) {
+                ((MapOperation) operation).afterRunFinal();
+            }
```

Comment:
I was wondering what if we introduce a method `afterRunFinal()` in Operation class with empty body, and it would always be invoked in the `finally` block. `afterRunFinal()` looks like a useful method to have, wdyt?

Reply:
It is a matter of taste :-) We have a plenty of operations not related to the IMap, tiered-store, etc. We need this method only for the tstore and this is why I wanted to minimize an impact of the change on other data structures.

Reply:
sounds logical, thanks :)